Godeps/_workspace/src/github.com/docker/spdystream/connection.go

```diff
@@ -317,9 +322,13 @@ func (s *Connection) Serve(newHandler StreamHandler) {
 			partition = partitionRoundRobin
 			partitionRoundRobin = (partitionRoundRobin + 1) % FRAME_WORKERS
 		case *spdy.GoAwayFrame:
-			priority = 0
-			partition = partitionRoundRobin
-			partitionRoundRobin = (partitionRoundRobin + 1) % FRAME_WORKERS
+			/*
+				priority = 0
+				partition = partitionRoundRobin
+				partitionRoundRobin = (partitionRoundRobin + 1) % FRAME_WORKERS
+			*/
+			s.handleGoAwayFrame(frame)
```

Comment:
It's probably ok to let a worker handle it. This way ensures that no additional frames can be processed from the conn.